packages/NodeTypeResolver/src/PerNodeTypeResolver/MethodCallTypeResolver.php

```diff
@@ -48,76 +55,70 @@ public function getNodeClasses(): array
      */
     public function resolve(Node $methodCallNode): array
     {
-        // 1. get $anotherVar type
-
-        /** @var Variable|mixed $variableNode */
-        $variableNode = $methodCallNode->var;
-
-        if (! $variableNode instanceof Variable) {
+        if ($methodCallNode->var instanceof MethodCall) {
+            // chain method calls: $this->someCall()->anotherCall()
+            $parentCallerTypes = $this->nodeTypeResolver->resolve($methodCallNode->var);
+        } elseif ($methodCallNode->var instanceof Variable) {
+            // $this->someCall()
+            $parentCallerTypes = $this->nodeTypeResolver->resolve($methodCallNode->var);
+        } elseif ($methodCallNode->var instanceof PropertyFetch) {
+            // $this->propertySomeCall()
+            $parentCallerTypes = $this->nodeTypeResolver->resolve($methodCallNode->var);
+        } else {
             return [];
         }
 
-        $variableName = (string) $variableNode->name;
-
-        $methodCallVariableTypes = $this->typeContext->getTypesForVariable($variableName);
-
-        $methodCallName = $this->resolveMethodCallName($methodCallNode);
-
-        if (! $methodCallVariableTypes || ! $methodCallName) {
+        $methodName = (string) $methodCallNode->name;
```

Comment:
@TomasVotruba I guess we will have problems with this like #236

Reply:
I see, we will. Could you send failing test to `master` for this?

Reply:
Sure, sending

Reply:
Oh, I thought you're reporting some error. In that case it's allright